### PR TITLE
Add extrinsic to share registry

### PR DESCRIPTION
**Pull request by linerocks**

### CODE REVIEW

File: pallets/meta-registry/src/lib.rs [hunk 1]
```diff
@@ -400,6 +436,46 @@ pub mod pallet {
...
+      );
+
+      if [AccessType::Issuer, AccessType::Accessor].contains(&author_access) {
+        ensure!(maybe_facility_id.is_some(), Error::<T>::FacilityIdRequired);
+
+        let facility_id = maybe_facility_id.clone().unwrap();
+
+        ensure!(facility_id == registry.facility_id, Error::<T>::FacilityIdMismatch);
+        ensure!(T::OrganizationRules::is_group_existed(&facility_id), Error::<T>::FacilityNotExisted);
+        ensure!(T::OrganizationRules::is_employee(accessor_id, &facility_id), Error::<T>::NotAuthorized);
```
Comment (altairrostam) [inline pallets/meta-registry/src/lib.rs:472]: This is for sharing record right? Does line 472 means that it can be shared if it's an employee? CMIIW

**Review by altairrostam — APPROVED** [review]

File: pallets/meta-registry/src/lib.rs [hunk 2]
```diff
@@ -252,6 +266,28 @@ pub mod pallet {
...
+
+    #[pallet::call_index(4)]
+    #[pallet::weight(10_000 + T::DbWeight::get().writes(1).ref_time())]
+    pub fn share_registry(
+      origin: OriginFor<T>,
+      registry_id: RegistryId,
+      accessor_id: T::AccountId,
+      access_type: AccessType,
+      maybe_facility_id: Option<T::AccountId>,
+    ) -> DispatchResult {
```
Comment (wilsonangara) [inline pallets/meta-registry/src/lib.rs:278]: what does it mean by `maybe_facility_id`?

**Review by wilsonangara — APPROVED** [review]

File: pallets/meta-registry/src/lib.rs (hunk 1 above)
Comment (linerocks) [reply pallets/meta-registry/src/lib.rs:472]: If the recipient is from the same organization, yes it is allowed

File: pallets/meta-registry/src/lib.rs (hunk 2 above)
Comment (linerocks) [reply pallets/meta-registry/src/lib.rs:278]: Rust strictly check for Option typed data. `maybe_` prefix signifies that the variables is still an Option, not yet unwrapped.

see about Option here: https://doc.rust-lang.org/std/option/

**Pull request merged**